src/IB/IBFEMethod.cpp

```diff
@@ -2871,6 +2898,49 @@ IBFEMethod::getScratchToPrimarySchedule(const int level_number,
     return *d_scratch_transfer_backward_schedules[key];
 } // getScratchToPrimarySchedule
 
+SAMRAI::xfer::RefineSchedule<NDIM>&
+IBFEMethod::getProlongationSchedule(const int level_number, const int coarse_data_idx, const int fine_data_idx)
+{
+    const auto key = std::make_pair(level_number, std::make_pair(coarse_data_idx, fine_data_idx));
+    if (d_prolongation_schedules.count(key) == 0)
+    {
+        Pointer<PatchHierarchy<NDIM> > hierarchy = d_use_scratch_hierarchy ? d_scratch_hierarchy : d_hierarchy;
+        Pointer<RefineAlgorithm<NDIM> > refine_algorithm = new RefineAlgorithm<NDIM>();
+        Pointer<RefineOperator<NDIM> > refine_op;
+
+        Pointer<hier::Variable<NDIM> > f_var;
+        VariableDatabase<NDIM>::getDatabase()->mapIndexToVariable(coarse_data_idx, f_var);
+        {
+            Pointer<hier::Variable<NDIM> > f_var_2;
+            VariableDatabase<NDIM>::getDatabase()->mapIndexToVariable(fine_data_idx, f_var_2);
+            // These should be the same variable
+            TBOX_ASSERT(&*f_var == &*f_var_2);
+        }
+
+        Pointer<CellVariable<NDIM, double> > f_cc_var = f_var;
+        Pointer<SideVariable<NDIM, double> > f_sc_var = f_var;
+        const bool cc_data = f_cc_var;
+        const bool sc_data = f_sc_var;
+        TBOX_ASSERT(cc_data || sc_data);
+        if (cc_data)
+        {
+            Pointer<CartesianGridGeometry<NDIM> > geometry = hierarchy->getGridGeometry();
+            refine_op = geometry->lookupRefineOperator(f_var, "CONSERVATIVE_LINEAR_REFINE");
+        }
+        else
+            refine_op = new CartSideDoubleRT0Refine();
+        TBOX_ASSERT(refine_op);
+        refine_algorithm->registerRefine(fine_data_idx, coarse_data_idx, fine_data_idx, refine_op);
+        Pointer<PatchLevel<NDIM> > fine_level = hierarchy->getPatchLevel(level_number + 1);
+        // We can ignore the fourth argument since we don't need to deal with
```

Comment:
Fifth argument?

Might be more clear to make a variable set to `nullptr` that is passed in as an argument to `createSchedule`.

I could imagine being confused by this when revisiting the code some day. 😀

Reply:
Oops - wrong comments are bad. Fixed.